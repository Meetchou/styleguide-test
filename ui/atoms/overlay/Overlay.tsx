import React, { FC } from 'react';
import cls from 'classnames';

export type OverlayType = {
  className?: string;
  onClick?: () => void
}

/**
 * Overlay content
 * @type {OverlayType}
 */
const Overlay: FC<OverlayType> = ({
  className,
  ...rest
}) => {

  const overlayCls = cls(
    'absolute top-0 right-0 left-0 w-full h-full bg-gray-900 opacity-40',
    className
  );

  return (
    <div className={overlayCls} {...rest}></div>
  );
};
export default Overlay;
