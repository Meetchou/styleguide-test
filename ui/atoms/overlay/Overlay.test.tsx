import React from 'react';

import render from '@ui/testing/render';

import Overlay from './Overlay';

describe('ui/atoms/overlay/Overlay', () => {
  describe('renders correctly', () => {
    it('normal mode', () => {
      const tree = render(<Overlay />);
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('className correctly passed', () => {
      const tree = render(<Overlay className='bg-white' />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

});
