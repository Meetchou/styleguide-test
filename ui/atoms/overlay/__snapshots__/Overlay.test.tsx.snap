// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/overlay/Overlay renders correctly className correctly passed 1`] = `
<div
  className="absolute top-0 right-0 left-0 w-full h-full bg-gray-900 opacity-40 bg-white"
/>
`;

exports[`ui/atoms/overlay/Overlay renders correctly normal mode 1`] = `
<div
  className="absolute top-0 right-0 left-0 w-full h-full bg-gray-900 opacity-40"
/>
`;
